flink-contrib/flink-streaming-contrib/src/main/java/org/apache/flink/contrib/streaming/state/DbBackendConfig.java

```diff
@@ -0,0 +1,406 @@
+	 */
+	public float getMaxKvCacheEvictFraction() {
+		return maxKvEvictFraction;
+	}
+
+	/**
+	 * The number of elements that will be evicted when the cache is full.
+	 * 
+	 */
+	public int getNumElementsToEvict() {
+		return (int) Math.ceil(getKvCacheSize() * getMaxKvCacheEvictFraction());
+	}
+
+	/**
+	 * Sets how often will automatic compaction be performed on the database to
+	 * remove old overwritten state changes. The frequency is set in terms of
+	 * number of successful checkpoints between two compactions and should take
+	 * the state size and checkpoint frequency into account.
+	 * <p>
+	 * By default automatic compaction is turned off.
+	 */
+	public void setKvStateCompactionFrequency(int compactEvery) {
+		this.kvStateCompactionFreq = compactEvery;
+	}
+
+	/**
+	 * Sets how often will automatic compaction be performed on the database to
+	 * remove old overwritten state changes. The frequency is set in terms of
+	 * number of successful checkpoints between two compactions and should take
+	 * the state size and checkpoint frequency into account.
+	 * <p>
+	 * By default automatic compaction is turned off.
+	 */
+	public int getKvStateCompactionFrequency() {
+		return kvStateCompactionFreq;
+	}
+
+	/**
+	 * The number of times each SQL command will be retried on failure.
+	 */
+	public int getMaxNumberOfSqlRetries() {
+		return maxNumberOfSqlRetries;
+	}
+
+	/**
+	 * Sets the number of times each SQL command will be retried on failure.
+	 */
+	public void setMaxNumberOfSqlRetries(int maxNumberOfSqlRetries) {
+		this.maxNumberOfSqlRetries = maxNumberOfSqlRetries;
+	}
+
+	/**
+	 * The number of milliseconds slept between two SQL retries. The actual
+	 * sleep time will be chosen randomly between 1 and the given time.
+	 * 
+	 */
+	public int getSleepBetweenSqlRetries() {
+		return sleepBetweenSqlRetries;
+	}
+
+	/**
+	 * Sets the number of milliseconds slept between two SQL retries. The actual
+	 * sleep time will be chosen randomly between 1 and the given time.
+	 * 
+	 */
+	public void setSleepBetweenSqlRetries(int sleepBetweenSqlRetries) {
+		this.sleepBetweenSqlRetries = sleepBetweenSqlRetries;
+	}
+
+	/**
+	 * Sets the partitioner used to assign keys to different database shards
+	 */
+	public void setPartitioner(Partitioner partitioner) {
+		this.shardPartitioner = partitioner;
+	}
+
+	/**
+	 * Creates a new {@link ShardedConnection} using the set parameters.
+	 * 
+	 * @throws SQLException
+	 */
+	public ShardedConnection createShardedConnection() throws SQLException {
+		if (JDBCDriver != null) {
+			try {
+				Class.forName(JDBCDriver);
+			} catch (ClassNotFoundException e) {
+				throw new RuntimeException("Could not load JDBC driver class", e);
+			}
+		}
+		if (shardPartitioner == null) {
+			return new ShardedConnection(shardUrls, userName, userPassword);
+		} else {
+			return new ShardedConnection(shardUrls, userName, userPassword, shardPartitioner);
+		}
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
Overriding equals() but not hashCode(). Not sure what the usage of this object is, but dropping the equals method may make sense (do you ever compare configs fro equality?).